<launch>
  <arg name="manager"             default="realsense_ros_camera_manager"/>
  <arg name="serial_no"           default="" />
  <arg name="camera"              default="camera"/>
  <arg name="depth"               default="depth" />
  <arg name="rgb"                 default="color" />
  <arg name="fisheye"             default="fisheye" />
  <arg name="accel"               default="accel" />
  <arg name="gyro"                default="gyro" />
  <arg name="fisheye_width"       default="640" />
  <arg name="fisheye_height"      default="480" />
  <arg name="enable_fisheye"      default="false" />
  <arg name="depth_width"         default="320" />
  <arg name="depth_height"        default="240" />
  <arg name="enable_depth"        default="true" />
  <arg name="color_width"         default="640" />
  <arg name="color_height"        default="480" />
  <arg name="enable_color"        default="true" />
  <arg name="fisheye_fps"         default="30" />
  <arg name="depth_fps"           default="30" />
  <arg name="color_fps"           default="30" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" output="screen"/> 
  <node pkg="nodelet" type="nodelet" name="realsense_ros_$(arg camera)" args="load realsense_ros_camera/NodeletCamera $(arg manager) --no-bond" output="screen">
    <param name="serial_no"                type="str"  value="$(arg serial_no)" />
    
    <param name="fisheye_width"            type="int"  value="$(arg fisheye_width)" />
    <param name="fisheye_height"           type="int"  value="$(arg fisheye_height)" />
    <param name="enable_fisheye"           type="bool" value="$(arg enable_fisheye)" />
    <param name="depth_width"              type="int"  value="$(arg depth_width)" />
    <param name="depth_height"             type="int"  value="$(arg depth_height)" />
    <param name="enable_depth"             type="bool" value="$(arg enable_depth)" />
    <param name="color_width"              type="int"  value="$(arg color_width)" />
    <param name="color_height"             type="int"  value="$(arg color_height)" />
    <param name="enable_color"             type="bool" value="$(arg enable_color)" />
    <param name="fisheye_fps"              type="int"  value="$(arg fisheye_fps)" />
    <param name="depth_fps"                type="int"  value="$(arg depth_fps)" />
    <param name="color_fps"                type="int"  value="$(arg color_fps)" />

    <param name="depth_optical_frame_id"   type="str"  value="$(arg camera)_depth_optical_frame" />
    <param name="color_optical_frame_id"   type="str"  value="$(arg camera)_rgb_optical_frame" />
    <param name="ir_optical_frame_id"      type="str"  value="$(arg camera)_ir_optical_frame" />
    <param name="ir2_optical_frame_id"     type="str"  value="$(arg camera)_ir2_optical_frame" />
    <param name="fisheye_optical_frame_id" type="str"  value="$(arg camera)_fisheye_optical_frame" />
    <param name="accel_optical_frame_id"   type="str"  value="$(arg camera)_accel_optical_frame" />
    <param name="gyro_optical_frame_id"    type="str"  value="$(arg camera)_gyro_optical_frame" />
  </node>
</launch>
